src/lazy/binary/encoded_value.rs

```diff
@@ -41,7 +41,7 @@ impl EncodedHeader for Header {
 /// without re-parsing its header information each time.
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub(crate) struct EncodedValue<HeaderType: EncodedHeader> {
-    pub(crate) encoding: ParameterEncoding,
+    pub(crate) encoding: BinaryValueEncoding,
```

Comment:
🗺️ Previously, `ParameterEncoding` was a `Copy` type:

https://github.com/amazon-ion/ion-rust/blob/6f04a8e3e9cbc62d56c7d46eb12563773b8baa08/src/lazy/expanded/template.rs#L82-L88

Starting with this PR (which stubs out macro-shaped parameter support), parameters needed the ability to hold a reference to a `Macro` that would last as long as necessary. To achieve this, a new variant was added: `MacroShaped(Rc<Macro>)`.

This meant that `ParameterEncoding` was no longer `Copy`. However, the only place it was used that required it to be `Copy` was the `EncodedValue` struct, which was using it as a proxy for "encoding backing this binary value". Since binary values can only be backed by _value_ encodings (not macros), it made sense to introduce a new type for that use case.